### PR TITLE
EDIT: Heavy duty endpoints

### PR DESCRIPTION
**Pull request by kodaline**
## Description

Memory, LLM/Embedder settings and Plugins now are sync and wait until operation is not performed.

Relates to issue #60 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/directives/vLock.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const vLock = {
+    mounted(el: { classList: { add: (arg0: string) => void; remove: (arg0: string) => void; }; }, binding: { value: any; }) {
+        if (binding.value) {
+            el.classList.add('vlock')
+        } else {
+            el.classList.remove('vlock')
+        }
+      },
+      updated(el: { _magicSpinnerWrapper: { style: { display: string; }; }; classList: { add: (arg0: string) => void; remove: (arg0: string) => void; }; }, binding: { value: any; }) {
```
Comment (zAlweNy26) [inline src/directives/vLock.ts:9]: I understand typing everything, but this is too much even for me 🤣

Comment (kodaline) [reply src/directives/vLock.ts:9]: Lol automatically done by VSC 😸 I will simplify it

File: src/App.vue [hunk 2]
```diff
@@ -2,7 +2,12 @@
 import { useSettings } from '@stores/useSettings'
 import ModalBox from '@components/ModalBox.vue'
 import { updateAuthKey } from '@/api'
+import useStoreMapping from '@/utils/storeRouteMapping'
+import vLock from '@/directives/vLock'
```
Comment (zAlweNy26) [inline src/App.vue:6]: Why did you import the directive? You already declared it globally in the `main.ts` file

File: src/main.ts [hunk 3]
```diff
@@ -23,8 +24,8 @@ pinia.use(({ store }) => {
 	const state = cloneDeep(store.$state)
 	store.$reset = () => store.$patch(state)
 })
-
 app.use(pinia)
 app.use(router)
+app.directive('vLock', vLock)
```
Comment (zAlweNy26) [inline src/main.ts:29]: The directive name should be just `lock`, Vue already adds the `v-` prefix when declared globally

**Comment by zAlweNy26** [comment]
Do you want to fix the last thing?

**Comment by kodaline** [comment]
> Do you want to fix the last thing?

Do you mean the one to add the CSS property directly to the directive hooks?

**Comment by zAlweNy26** [comment]
> > Do you want to fix the last thing?
> 
> Do you mean the one to add the CSS property directly to the directive hooks?

Yes

**Pull request merged**